cli/flox-rust-sdk/src/models/environment/remote_environment.rs

```diff
@@ -79,20 +86,35 @@ impl RemoteEnvironment {
         )
         .unwrap();
 
-        let out_link = gcroots_dir(flox, &pointer.owner).join(remote_branch_name(&pointer));
+        let inner_out_link = gcroots_dir(flox, &pointer.owner).join(remote_branch_name(&pointer));
+        let mut inner =
+            ManagedEnvironment::open_with(floxmeta, flox, pointer, dot_flox_path, inner_out_link)
+                .map_err(RemoteEnvironmentError::OpenManagedEnvironment)?;
 
-        let inner = ManagedEnvironment::open_with(floxmeta, flox, pointer, dot_flox_path, out_link)
-            .map_err(RemoteEnvironmentError::OpenManagedEnvironment)?;
+        inner
+            .pull(true)
+            .map_err(RemoteEnvironmentError::ResetManagedEnvironment)?;
+
+        let out_link = {
+            let link_path = path.join(GCROOTS_DIR_NAME);
```

Comment:
question blocking: should we still have system in this path? Looks like we have it for path but not managed environments?

Reply:
Let's discuss this separately